pontoon/base/models.py

```diff
@@ -891,7 +892,11 @@ def save(self, imported=False, *args, **kwargs):
                 .filter(entity=self.entity, locale=self.locale, plural_form=self.plural_form)
                 .exclude(pk=self.pk)
                 .update(approved=False, approved_user=None, approved_date=None))
-
+            if not self.memory_entries.all().exists():
```

Comment:
You can ommit .all().

We also need to update the TranslationMemoryEntry table on sync, when we bulk save/update translations, which skips the custom save() method I think.